src/V3SplitVar.cpp

```diff
@@ -228,6 +228,25 @@ const char* const SplitVarImpl::notSplitMsg
 // AstVarRef    -> Create a temporary variable and refer the variable
 // AstSliceSel  -> Create a temporary variable and refer the variable
 
+// Compare AstNode* to get deterministic ordering when showing messages.
+struct AstNodeComparator {
+    bool operator()(const AstNode* ap, const AstNode* bp) const {
+        const FileLine* afp = ap->fileline();
+        const FileLine* bfp = bp->fileline();
+        if (afp->firstLineno() != bfp->firstLineno())
+            return afp->firstLineno() < bfp->firstLineno();
+        if (afp->firstColumn() != bfp->firstColumn())
+            return afp->firstColumn() < bfp->firstColumn();
+        // Don't comapre its filename because it is expensive.
+        // Line number and column is practically enough.
+        // The comparison of raw pointer here is just in case.
+        // The comparison of this pointer may differ on different run,
+        // but this is just warning message order. (mostly for CI)
+        // Verilated result is same anyway.
+        return ap < bp;
```

Comment:
Suggest make an operatorCompare() in V3FileLine.h near the operator== and compare that.  Note filenames are stored as numbers so you can compare filenameno (a number).  When we get to C++20 this can become the spaceship operator.

Reply:
Thanks for the info.
Then the comparison function can check m_filenameno first so that the sorted result would be natural. (filename -> line no. -> column no)

Let me confirm the signature. Do you mean 1. below ?
1. bool FileLine::operatorCompare(const FileLine&) const;
2. bool FileLine::operator < (const FileLine&) const;

Reply:
I think you need a operatorCompare as need -1, 0 and 1.

Reply:
If you can stable sort with just operator< then that's better instead.

Reply:
You are right. If I just sort FileLine itself, `operator <` is enough.
But I'd like to also know if two FileLine are same or not.
So returning +-1, 0 is more convenient.
Added in 00ff30b5ad712aa74e6bff5a3c67154195deb820.